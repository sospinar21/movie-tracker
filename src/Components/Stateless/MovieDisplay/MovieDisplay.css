.movies-container {
  background-color: #fffdfd;
  box-shadow: 2px 0px 20px #cecccc;
  display: inline-flex;
  height: 100%;
  justify-content: space-between;
  margin-left: 4.5%;
  overflow-x: scroll;
  padding: 4vh 1vh 0;
  text-align: center;
  width: 90%; }

.movie-card, .favorite-card p {
  color: white;
  font-size: 18px;
  margin: 0 auto 10px; }

.movie-card {
  background-color: black;
  box-shadow: 2px 0px 20px #cecccc;
  box-sizing: content-box;
  margin: 0px 20px 25px;
  padding: 4vh;
  width: 35vh; }

.arrows {
  width: 100%;
  position: absolute;
  z-index: -1;
  margin-top: 15%;
  display: flex;
  justify-content: space-between; }

svg {
  fill: red; }

.left-arrow {
  width: 7vh;
  background-size: cover;
  transform: rotate(180deg); }

.right-arrow {
  width: 7vh;
  background-size: cover; }

.movie-card img {
  width: inherit; }

.favorite-card {
  background-color: #6d0404;
  box-shadow: 2px 0px 20px #cecccc;
  box-sizing: content-box;
  margin: 0px 20px 25px;
  padding: 4vh;
  width: 35vh; }

.favorite-card img {
  width: inherit; }

.scroll::-webkit-scrollbar {
  width: 10px; }

.scroll::-webkit-scrollbar-track {
  -webkit-box-shadow: inset 0 0 6px #535151;
  border-radius: 10px; }

.scroll::-webkit-scrollbar-thumb {
  background-color: #fabf52;
  border-radius: 10px;
  outline: 2px solid #000000; }
