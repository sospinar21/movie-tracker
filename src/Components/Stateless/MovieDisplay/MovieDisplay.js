import React, { Component } from 'react';
import {fetchMovieData, postMovieToFavorites, retrieveFavorites} from './../../../Helpers/apiCalls';
import './MovieDisplay.css';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { addMovies, addFavorites } from './../../../Actions/actions';

class MovieDisplay extends Component {
  constructor() {
    super();
  }

  async componentDidMount () {
    const movieData = await fetchMovieData(this.props.user);
    this.props.addMovies(movieData);
  }

  async updateFavorites (movie) {
    const userFavorites = await retrieveFavorites(this.props.user.id);
    this.props.addFavorites(userFavorites);
    postMovieToFavorites(movie, this.props.favorites);
  }
  
  render () {

    const movieDisplay = this.props.movies.map((movie, index) => {
      let newMovie = {...movie};
      
      return (
        <div key={index} className='movie-card' >
          <p>{movie.title}</p>
          <img src={`https://image.tmdb.org/t/p/w500${movie.poster_path}`} alt={movie.title}/>
          <button onClick={() => this.updateFavorites(movie)}>Favorite</button>
        </div>
      );
    });

    return (
      <div>
        <div className='arrows'>
          <div className='left-arrow'>
            <svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24'>
              <path d='M0 0h24v24H0z' fill='none'/>
              <path d='M10 16.5l6-4.5-6-4.5v9zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z'/>
            </svg> 
          </div>      
          <div className='right-arrow' >
            <svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24'>
              <path d='M0 0h24v24H0z' fill='none'/>
              <path d='M10 16.5l6-4.5-6-4.5v9zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z'/>
            </svg>
          </div>
        </div>
        <div className='movies-container scroll'>
          { movieDisplay }
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return ({
    movies: state.movies,
    favorites: state.favorites,
    user: state.user
  });
};

const mapDispatchToProps = dispatch => ({
  addMovies: (movieData) => dispatch(addMovies(movieData)),
  addFavorites: (movie) => dispatch(addFavorites(movie))
});

MovieDisplay.propTypes = {
  movies: PropTypes.array,
  addMovies: PropTypes.func,
  addFavorites: PropTypes.func
};

export default connect(mapStateToProps, mapDispatchToProps)(MovieDisplay);