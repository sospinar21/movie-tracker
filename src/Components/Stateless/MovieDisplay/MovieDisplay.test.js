import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import React from 'react';
import { MovieDisplay, mapStateToProps, mapDispatchToProps } from './MovieDisplay';
import  fetchMovieData  from '../../../Helpers/apiCalls';

describe('movieDisplay', () => {
  
  let movieDisplay;
  let mockProps;
  window.fetch = jest.fn();

  beforeEach(() => {

    mockProps = {
      user: {},
      favorites: [{favoriteId: 1}],
      movies: [{movie_id: 4, title: 'Spiderman'}, {movie_id: 1, title:'Batman'}],
      removeFavorite: jest.fn(),
      addMovies: jest.fn(),
    };

    movieDisplay = shallow(<MovieDisplay {...mockProps} />, { disableLifecycleMethods: true });

    window.fetch.mockImplementation(() => Promise.resolve({
      status: 200,
      json: () => Promise.resolve({data: {}})
    }));
  });

  describe('component', () => {

    it('matches snapshot', () => {
      expect(movieDisplay).toMatchSnapshot();
    });
  })

  describe('handleSelectedMovie', () => {

    it('sets state with the new selected movie', () => {

      const movie = {movie_id: 1, title:'Batman'}

      movieDisplay.instance().handleSelectedMovie(movie)

      expect(movieDisplay.state('selectedMovie')).toEqual(movie)
      
    });
  })

  describe('updateFavorites', () => {

    it('calls remove favorites with the correct params', () => {

      const movie = {movie_id: 1, title:'Batman'}

      movieDisplay.instance().updateFavorites(movie)

      expect(mockProps.removeFavorite).toHaveBeenCalledWith(movie)

    });
  })

  describe('mapStateToProps', () => {

    it('should return an object with user, favorites and movies', () => {

      const mockState = {
        user: {name: 'Tony'},
        favorites: [],
        movies: []
      };

      const expected = {
        user: {name: 'Tony'},
        favorites: [],
        movies: []
      };

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);

    });
  });

  describe('mapDispatchToProps', () => {

    let mockDispatch;
    let movie;
    let mappedProps

    beforeEach(() => {

      mockDispatch = jest.fn();
      movie = {movie_id: 1, title:'Batman'}
      mappedProps = mapDispatchToProps(mockDispatch);
    })

    it('should call dispatch with the correct params', () => {

      const mockAction = {
        type: 'REMOVE_FAVORITE',
        movie
      };
      
      mappedProps.removeFavorite(movie);
      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    })

    it('should call dispatch with the add movie action', () => {

      const movieData = {movie_id: 1, title:'Batman'}

      const mockAction = {
        type: 'ADD_MOVIES',
        movieData
      };
      
      mappedProps.addMovies(movie);
      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    })

    it('should call dispatch with the add movie action', () => {

      const movies = {movie_id: 1, title:'Batman'}

      const mockAction = {
        type: 'ADD_FAVORITE',
        movies
      };
      
      mappedProps.addFavorites(movie);
      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    })


  })

})