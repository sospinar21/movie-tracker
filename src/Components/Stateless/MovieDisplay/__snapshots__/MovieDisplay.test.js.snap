// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`movieDisplay component matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieDisplay
    addMovies={[Function]}
    favorites={
        Array [
            Object {
              "favoriteId": 1,
            },
          ]
    }
    movies={
        Array [
            Object {
              "movie_id": 4,
              "title": "Spiderman",
            },
            Object {
              "movie_id": 1,
              "title": "Batman",
            },
          ]
    }
    removeFavorite={[Function]}
    user={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="selected-movie"
>
          <div
                    className="pick-movie-text"
          >
                    <p>
                              Select a movie
                    </p>
          </div>
</div>,
        <div
          className="movies-container scroll"
>
          <div
                    className="movie-card"
          >
                    <img
                              alt="Spiderman"
                              className="movie-poster"
                              onClick={[Function]}
                              src="https://image.tmdb.org/t/p/w500undefined"
                    />
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="nav-favorite"
                              to="/signup"
                    >
                              <button
                                        className="fav-button"
                              >
                                        ☆
                              </button>
                    </NavLink>
          </div>
          <div
                    className="movie-card"
          >
                    <img
                              alt="Batman"
                              className="movie-poster"
                              onClick={[Function]}
                              src="https://image.tmdb.org/t/p/w500undefined"
                    />
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="nav-favorite"
                              to="/signup"
                    >
                              <button
                                        className="fav-button-selected"
                              >
                                        ★
                              </button>
                    </NavLink>
          </div>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="pick-movie-text"
>
            <p>
                        Select a movie
            </p>
</div>,
          "className": "selected-movie",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p>
              Select a movie
</p>,
            "className": "pick-movie-text",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Select a movie",
            },
            "ref": null,
            "rendered": "Select a movie",
            "type": "p",
          },
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="movie-card"
>
              <img
                            alt="Spiderman"
                            className="movie-poster"
                            onClick={[Function]}
                            src="https://image.tmdb.org/t/p/w500undefined"
              />
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            className="nav-favorite"
                            to="/signup"
              >
                            <button
                                          className="fav-button"
                            >
                                          ☆
                            </button>
              </NavLink>
</div>,
            <div
              className="movie-card"
>
              <img
                            alt="Batman"
                            className="movie-poster"
                            onClick={[Function]}
                            src="https://image.tmdb.org/t/p/w500undefined"
              />
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            className="nav-favorite"
                            to="/signup"
              >
                            <button
                                          className="fav-button-selected"
                            >
                                          ★
                            </button>
              </NavLink>
</div>,
          ],
          "className": "movies-container scroll",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="Spiderman"
                  className="movie-poster"
                  onClick={[Function]}
                  src="https://image.tmdb.org/t/p/w500undefined"
/>,
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="nav-favorite"
                  to="/signup"
>
                  <button
                                    className="fav-button"
                  >
                                    ☆
                  </button>
</NavLink>,
              ],
              "className": "movie-card",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Spiderman",
                  "className": "movie-poster",
                  "onClick": [Function],
                  "src": "https://image.tmdb.org/t/p/w500undefined",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": <button
                    className="fav-button"
>
                    ☆
</button>,
                  "className": "nav-favorite",
                  "to": "/signup",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "☆",
                    "className": "fav-button",
                  },
                  "ref": null,
                  "rendered": "☆",
                  "type": "button",
                },
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="Batman"
                  className="movie-poster"
                  onClick={[Function]}
                  src="https://image.tmdb.org/t/p/w500undefined"
/>,
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="nav-favorite"
                  to="/signup"
>
                  <button
                                    className="fav-button-selected"
                  >
                                    ★
                  </button>
</NavLink>,
              ],
              "className": "movie-card",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Batman",
                  "className": "movie-poster",
                  "onClick": [Function],
                  "src": "https://image.tmdb.org/t/p/w500undefined",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": <button
                    className="fav-button-selected"
>
                    ★
</button>,
                  "className": "nav-favorite",
                  "to": "/signup",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "★",
                    "className": "fav-button-selected",
                  },
                  "ref": null,
                  "rendered": "★",
                  "type": "button",
                },
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="selected-movie"
>
            <div
                        className="pick-movie-text"
            >
                        <p>
                                    Select a movie
                        </p>
            </div>
</div>,
          <div
            className="movies-container scroll"
>
            <div
                        className="movie-card"
            >
                        <img
                                    alt="Spiderman"
                                    className="movie-poster"
                                    onClick={[Function]}
                                    src="https://image.tmdb.org/t/p/w500undefined"
                        />
                        <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    className="nav-favorite"
                                    to="/signup"
                        >
                                    <button
                                                className="fav-button"
                                    >
                                                ☆
                                    </button>
                        </NavLink>
            </div>
            <div
                        className="movie-card"
            >
                        <img
                                    alt="Batman"
                                    className="movie-poster"
                                    onClick={[Function]}
                                    src="https://image.tmdb.org/t/p/w500undefined"
                        />
                        <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    className="nav-favorite"
                                    to="/signup"
                        >
                                    <button
                                                className="fav-button-selected"
                                    >
                                                ★
                                    </button>
                        </NavLink>
            </div>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="pick-movie-text"
>
              <p>
                            Select a movie
              </p>
</div>,
            "className": "selected-movie",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p>
                Select a movie
</p>,
              "className": "pick-movie-text",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Select a movie",
              },
              "ref": null,
              "rendered": "Select a movie",
              "type": "p",
            },
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="movie-card"
>
                <img
                                alt="Spiderman"
                                className="movie-poster"
                                onClick={[Function]}
                                src="https://image.tmdb.org/t/p/w500undefined"
                />
                <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                className="nav-favorite"
                                to="/signup"
                >
                                <button
                                                className="fav-button"
                                >
                                                ☆
                                </button>
                </NavLink>
</div>,
              <div
                className="movie-card"
>
                <img
                                alt="Batman"
                                className="movie-poster"
                                onClick={[Function]}
                                src="https://image.tmdb.org/t/p/w500undefined"
                />
                <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                className="nav-favorite"
                                to="/signup"
                >
                                <button
                                                className="fav-button-selected"
                                >
                                                ★
                                </button>
                </NavLink>
</div>,
            ],
            "className": "movies-container scroll",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    alt="Spiderman"
                    className="movie-poster"
                    onClick={[Function]}
                    src="https://image.tmdb.org/t/p/w500undefined"
/>,
                  <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav-favorite"
                    to="/signup"
>
                    <button
                                        className="fav-button"
                    >
                                        ☆
                    </button>
</NavLink>,
                ],
                "className": "movie-card",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "Spiderman",
                    "className": "movie-poster",
                    "onClick": [Function],
                    "src": "https://image.tmdb.org/t/p/w500undefined",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "ariaCurrent": "true",
                    "children": <button
                      className="fav-button"
>
                      ☆
</button>,
                    "className": "nav-favorite",
                    "to": "/signup",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "☆",
                      "className": "fav-button",
                    },
                    "ref": null,
                    "rendered": "☆",
                    "type": "button",
                  },
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    alt="Batman"
                    className="movie-poster"
                    onClick={[Function]}
                    src="https://image.tmdb.org/t/p/w500undefined"
/>,
                  <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav-favorite"
                    to="/signup"
>
                    <button
                                        className="fav-button-selected"
                    >
                                        ★
                    </button>
</NavLink>,
                ],
                "className": "movie-card",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "Batman",
                    "className": "movie-poster",
                    "onClick": [Function],
                    "src": "https://image.tmdb.org/t/p/w500undefined",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "ariaCurrent": "true",
                    "children": <button
                      className="fav-button-selected"
>
                      ★
</button>,
                    "className": "nav-favorite",
                    "to": "/signup",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "★",
                      "className": "fav-button-selected",
                    },
                    "ref": null,
                    "rendered": "★",
                    "type": "button",
                  },
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
